src/openzeppelin/tests/test_account.cairo

```diff
@@ -143,55 +147,81 @@ fn test_is_valid_signature() {
 #[test]
 #[available_gas(2000000)]
 fn test_validate_deploy() {
-    let class_hash = 0x123;
-    let salt = 0;
-    let public_key = 0x234;
     let account = setup_dispatcher(Option::Some(@SIGNED_TX_DATA()));
 
     assert(
-        account.__validate_deploy__(class_hash, salt, public_key) == starknet::VALIDATED,
-        'Should validate correctly'
+        account.__validate_deploy__(0, 0, 0) == starknet::VALIDATED, 'Should validate correctly'
     );
 }
 
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Account: invalid signature', 'ENTRYPOINT_FAILED'))]
+fn test_validate_deploy_invalid() {
+    let mut data = SIGNED_TX_DATA();
+    data.transaction_hash += 1;
+    let account = setup_dispatcher(Option::Some(@data));
+
+    account.__validate_deploy__(0, 0, 0);
+}
+
 #[test]
 #[available_gas(2000000)]
 fn test_validate_declare() {
-    let class_hash = 0x123;
     let account = setup_dispatcher(Option::Some(@SIGNED_TX_DATA()));
 
-    assert(
-        account.__validate_declare__(class_hash) == starknet::VALIDATED, 'Should validate correctly'
-    );
+    assert(account.__validate_declare__(0) == starknet::VALIDATED, 'Should validate correctly');
 }
 
 #[test]
 #[available_gas(2000000)]
-fn test_execute() {
+#[should_panic(expected: ('Account: invalid signature', 'ENTRYPOINT_FAILED'))]
+fn test_validate_declare_invalid() {
+    let mut data = SIGNED_TX_DATA();
+    data.transaction_hash += 1;
+    let account = setup_dispatcher(Option::Some(@data));
+
+    account.__validate_declare__(0);
+}
+
+fn test_execute_with_version(version: Option<felt252>) {
     let data = SIGNED_TX_DATA();
     let account = setup_dispatcher(Option::Some(@data));
     let initial_public_key = data.public_key;
     let mut calls = ArrayTrait::new();
 
-    let set_public_key_selector = 0x2e3e21ff5952b2531241e37999d9c4c8b3034cccc89a202a6bf019bdf5294f9;
-
-    assert(account.get_public_key() == initial_public_key, 'Should get initial public key');
-
     let mut calldata = ArrayTrait::new();
-    let new_public_key = 0x789789;
-    calldata.append(new_public_key);
+    calldata.append(NEW_PUBKEY);
     let call = Call {
-        to: account.contract_address, selector: set_public_key_selector, calldata: calldata
+        to: account.contract_address, selector: SET_PUBLIC_KEY_SELECTOR, calldata: calldata
     };
-
     calls.append(call);
+
+    if version.is_some() {
+        testing::set_version(version.unwrap());
+    }
     let ret = account.__execute__(calls);
 
-    assert(account.get_public_key() == 0x789789, 'Should get new public key');
+    assert(account.get_public_key() == NEW_PUBKEY, 'Should get new public key');
+}
 
-    // Test return value
-    let mut call_retval = ret.at(0).span();
-    assert(call_retval.len() == 0, 'Should be an empty response');
+#[test]
+#[available_gas(2000000)]
+fn test_execute() {
+    test_execute_with_version(Option::None(()));
```

Comment:
what about moving the `setup_dispatcher` version setting in here? or do we need it in other tests as well?

Reply:
The version is part of the context that setup_dispatcher is initializing to make the calls in tests closer to reality. Is true that we can test validate_deploy (for ex.) without setting the correct transaction version, but IMO this is more error-prone because we can easier forget to set this context on a test that passes because of that. I think setting the basic context in setup_dispatcher, and using setup_dispatcher when possible is a better pattern than direct calls with incorrect context (as before to mention that there is a line, but for me most tests should use setup_dispatcher instead of direct calls)